### PR TITLE
One tag only feature

### PR DESCRIPTION
**Pull request by brumik**
One tag only mode allows only one tag from all categories by deleting all of the previously added tags.

* Added only one more reducer: `deleteAllAssignedTags`
* Updated tests with the new prop
* Fixed up the linter errors in the files which I touched (Mainly in the `tagging.stories.js` and `tagging.tests.js`)

@skateman @PanSpagetka 

Screenshot from 2019-07-04 11-52-24

### CODE REVIEW

**Comment by brumik** [comment]
If this is OK can we merge? @PanSpagetka ?

**Comment by skateman** [comment]
Gonna take a look too, gimme a minute...

**Comment by skateman** [comment]
btw @brumik a screenshot would be nice in the description

**Comment by skateman** [comment]
maybe it's a UX nitpick and @terezanovotna would say no, but I'd like to have both dropdowns "stuck" on the selected tag item if possible, having just one of them is confusing

**Comment by brumik** [comment]
I don't understand your last comment, what you mean?

For the screenshot, the UI is the same that for the other tags (and there is a story in the storybook for it)

**Comment by skateman** [comment]
Dropdown on the left after selecting a tag:
Screenshot from 2019-07-04 11-54-09

Upper one is stuck, the lower one is reset...IMO both should be the same value as it has been selected.

**Comment by brumik** [comment]
I corrected that bug here: #132 which is forked from this branch

**Comment by brumik** [comment]
With the styling:
Screenshot from 2019-07-04 12-59-30
Screenshot from 2019-07-04 12-59-43


Without it:
Screenshot from 2019-07-04 13-00-27
Screenshot from 2019-07-04 13-00-39


Note also the small X-es next to multiselect. If the UI implements it, it is working properly.

**Comment by skateman** [comment]
Oh, I can live with that one!

**Comment by brumik** [comment]
@Hyperkid123 Merge?

**Comment by brumik** [comment]
@miq-bot add_reviewer @Hyperkid123

File: src/tagging/reducers/reducers.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import * as actionsConstants from '../actions/actions';
-// state = state.assignedTags
+
+/* ================== SECTION: state = state.assignedTags ================== */
```
Comment (Hyperkid123) [inline src/tagging/reducers/reducers.js]: Can we rather use JS DOC notation instead of these comments?

Comment (brumik) [reply src/tagging/reducers/reducers.js]: Yes, but I didn't found how to make a section of code.

File: src/tagging/components/Tagging/Tagging.js [hunk 2]
```diff
@@ -13,6 +13,13 @@ class Tagging extends React.Component {
       tagCategory: this.props.selectedTagCategory,
       tagValue: selectedTagValue,
     };
+
+    // Single Tag Only
+    if (this.props.options && this.props.options.onlySingleTag) {
+      this.props.onSingleTagValueChange(action);
+      return;
```
Comment (karelhala) [inline src/tagging/components/Tagging/Tagging.js]: No need to `return;` but rewrite it to use else if
```JS
if (this.props.options && this.props.options.onlySingleTag) {
  this.props.onSingleTagValueChange(action);
} else if (this.findSelectedTag(this.props.selectedTagCategory).singleValue) {
  this.props.onTagValueChange(action, this.props.options);
}
```

**Pull request merged**